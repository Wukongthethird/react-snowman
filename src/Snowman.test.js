import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Snowman from "./Snowman"

it("Game finishes after nWrong reaches MaxGuess", function(){
  const { container } = render(<Snowman maxWrong = {1}/>);

  const button = container.querySelector( 'button[value="j"]' )
  fireEvent.click(button)
  expect(container).toContainHTML("You Lose")
  expect(container).toContainHTML("apple")
  expect(container).toMatchSnapshot()

  // const rightArrow = container.querySelector(".fa-chevron-circle-right");
  // fireEvent.click(rightArrow);

  // // expect the second image to show, but not the first
  // expect(
  //   container.querySelector('img[alt="Photo by Richard Pasquarella on Unsplash"]')
  // ).not.toBeInTheDocument();
  // expect(
  //   container.querySelector('img[alt="Photo by Pratik Patel on Unsplash"]')
  // ).toBeInTheDocument();

  // expect(container).toMatchSnapshot();


} )