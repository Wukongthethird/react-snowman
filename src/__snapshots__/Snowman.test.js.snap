// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game finishes after nWrong reaches MaxGuess 1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="1"
      src="1.png"
    />
    <p>
       Wrong: 
      1
    </p>
    <p
      class="Snowman-word"
    >
       
      apple
       
    </p>
    <p>
      You Lose
       
    </p>
  </div>
</div>
`;
